strawberryfields/backends/bosonicbackend/backend.py

```diff
@@ -277,39 +279,187 @@ def reset(self, pure=True, **kwargs):
         self.circuit.reset(self._init_modes)
 
     def prepare_thermal_state(self, nbar, mode):
-        self.circuit.init_thermal(nbar, mode)
+        # self.circuit.init_thermal(nbar, mode)
+        pass
 
     def prepare_vacuum_state(self, mode):
-        self.circuit.loss(0.0, mode)
+        # self.circuit.loss(0.0, mode)
+        pass
 
     def prepare_coherent_state(self, r, phi, mode):
-        self.circuit.loss(0.0, mode)
-        self.circuit.displace(r, phi, mode)
+        # self.circuit.loss(0.0, mode)
+        # self.circuit.displace(r, phi, mode)
+        pass
 
     def prepare_squeezed_state(self, r, phi, mode):
-        self.circuit.loss(0.0, mode)
-        self.circuit.squeeze(r, phi, mode)
+        # self.circuit.loss(0.0, mode)
+        # self.circuit.squeeze(r, phi, mode)
+        pass
 
     def prepare_displaced_squeezed_state(self, r_d, phi_d, r_s, phi_s, mode):
-        self.circuit.loss(0.0, mode)
-        self.circuit.squeeze(r_s, phi_s, mode)
-        self.circuit.displace(r_d, phi_d, mode)
+        # self.circuit.loss(0.0, mode)
+        # self.circuit.squeeze(r_s, phi_s, mode)
+        # self.circuit.displace(r_d, phi_d, mode)
+        pass
 
-    def prepare_cat(self, alpha, phi, desc):
+    def prepare_cat(self, alpha, phi, desc="complex"):
         """ Prepares the arrays of weights, means and covs for a cat state"""
-        return
+
+        # case alpha = 0 -> prepare vacuum
+        if np.isclose(np.absolute(alpha), 0):
+            return (
+                np.array([[1]], dtype=complex),
+                np.array([[0, 0]], dtype=complex),
+                np.array([[[0.5 * self.circuit.hbar, 0], [0, 0.5 * self.circuit.hbar]]]),
```

Comment:
same here: maybe better to write `np.array([0.5*self.circuit.hbar*np.identity(2)])`